src/app/heroes/heroes.service.ts

```diff
@@ -1,17 +1,34 @@
 import { Injectable } from '@angular/core';
-import { HttpClient } from '@angular/common/http';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/map';
 
 import { Hero } from '../models/hero';
 
 @Injectable()
 export class HeroesService {
+  private params: HttpParams = new HttpParams();
```

Comment:
Why do you use global property for entire class? Just use params variable in methods where it's needed.

Reply:
Fixed